shared/functions.js

```diff
@@ -5,6 +5,12 @@ import multihash from 'multihashes'
 import nacl from 'tweetnacl'
 import blake from 'blakejs'
 
+// Makes the `Buffer` global available in the browser.
+if (typeof window === 'object') {
+  const { Buffer } = require('buffer')
```

Comment:
Remember, per previous review, need to check to see if `Buffer` is already loaded.

Reply:
I think the `require()` call already does that internally, and will not load a given module twice unless its cache has been altered.

Reply:
@snowteamer I'm not so sure. The thing is `Buffer` can be defined by something other `require('buffer')`. For example, the way we previously included it is via `'rollup-plugin-node-globals'`, so in that case I don't think `require` would catch it. Think you need to check whether `Buffer` already exists, e.g. via `globals.Buffer` or `window.Buffer` or `typeof Buffer !== 'undefined'` (that last one might make the most sense)

Reply:
Ok, it makes sense in that case

Reply:
@taoeffect Is the following snippet suitable?
```js
// Makes the `Buffer` global available in the browser if needed.
if (typeof window === 'object' && !window.Buffer) {
  if (typeof Buffer === 'undefined')
    // Only import `Buffer` to hopefully help treeshaking.
    const { Buffer } = require('buffer')
    window.Buffer = Buffer
  } else {
    window.Buffer = Buffer
  }
}
```

Reply:
@snowteamer Looks ok, can be shortened I think:

```js
// Makes the `Buffer` global available in the browser if needed.
if (typeof window === 'object' && !window.Buffer && typeof Buffer === 'undefined') {
  window.Buffer = require('buffer').Buffer
}
```

The shorter the better (usually) :)

Reply:
Sure! :smile:
I was thinking about the case where the global `Buffer` has been defined elsewhere but not yet assigned to `window`; however in a browser context, that assignment would have already been performed by the browser itself I think.

So if you don't mind, I will just keep the `const Buffer = ...` declaration in case it helps treeshaking in the future (that would be nice because the `buffer` module is actually quite big, yielding more than 1k LOC in `dist/main.js`).

Reply:
@snowteamer Yeah np 👍